src/routes/productRoutes.js

```diff
@@ -1,20 +1,21 @@
 import controller from "../controllers/index.js";
 import validationSchemas from "../validations/index.js";
-import { validator } from "../middleware/validatorMW.js";
-
-const options = { warnings: true };
+import middleware from "../middleware/index.js";
 
 export function routes_Products(app) {
+  app.use("/product?s", middleware.authenticateUser);
```

Comment:
Why we are using `?`

Reply:
to match either product or products endpoint.